cunumeric/linalg/linalg.py

```diff
@@ -555,3 +576,19 @@ def _cholesky(a: ndarray, no_tril: bool = False) -> ndarray:
     )
     output._thunk.cholesky(input._thunk, no_tril=no_tril)
     return output
+
+
+def _solve(a: ndarray, b: ndarray) -> ndarray:
+    a_input = a
```

Comment:
is there a reason you want to make a copy?

Reply:
to convert the input arrays to float64.